frontend/src/metabase/models/containers/ModelDetailPage/ModelDetailPage.tsx

```diff
@@ -94,6 +100,26 @@ function ModelDetailPage({
     }
   }, [mainTable, hasFetchedTableMetadata, fetchTableForeignKeys]);
 
+  const handleNameChange = useCallback(
+    name => {
+      if (name && name !== model.displayName()) {
+        const nextCard = model.setDisplayName(name).card();
+        onChangeModel(nextCard as Card);
```

Comment:
Can we do something to avoid casting here?

Reply:
This is tricky because `Question` still uses the `Card` type coming from `metabase-types/types` and it can also be an unsaved card (i.e. without `id` which is often required here). I'm afraid we might need to make `metabase-lib` use up-to-date types first before this can be solved in a non-ugly way